File: src/main/java/ru/xpendence/topjavagraduation/entity/Dish.java
package ru.xpendence.topjavagraduation.entity;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.Setter;

import java.math.BigDecimal;

@Entity
@Table(name = "dishes")
@Getter
@Setter
public class  Dish {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "name")
    private String name;

    @Column(name = "price")
    private BigDecimal price;

    @Column(name = "active")
    private Boolean active;

    @ManyToOne(fetch = FetchType.LAZY, cascade = CascadeType.MERGE)
    @JoinColumn(name = "restaurant_id", nullable = false)
    private Restaurant restaurant;

    public static void enrichForUpdate(Dish forUpdate, Dish stored) {
        stored.name = forUpdate.name;
        stored.price = forUpdate.price;
        stored.active = forUpdate.active;
    }
}

File: src/main/kotlin/ru/xpendence/topjavagraduation/entity/Dish.kt
package ru.xpendence.topjavagraduation.entity

import java.math.BigDecimal

/**
 * Блюдо.
 */
data class Dish(
    val id: Long? = null,
    val name: String? = null,
    val price: BigDecimal? = null,
    val active: Boolean? = null,
    val restaurant: Restaurant,
) {
    companion object {
        @JvmStatic
        fun enrichForUpdate(forUpdate: Dish, stored: Dish): Dish =
            stored.copy(
                name = forUpdate.name,
                price = forUpdate.price,
                active = forUpdate.active,
            )
    }
}
